### PR TITLE
ci: github workflows

### PR DESCRIPTION
**Pull request by averen**
We should use GitHub workflows to continuously verify and deploy our codebase. This also allows others to contribute and improve checks or tests as necessary.

### CODE REVIEW

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Deploy to production
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  test-build:
+    name: Reusable workflow
```
Comment (averen) [inline .github/workflows/deploy-production.yml]: Need a better name

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: Pull Request Validation
+on: pull_request
+
+jobs:
+  test-build:
+    name: Reusable workflow
```
Comment (averen) [inline .github/workflows/pull-request.yml]: Need a better name

File: .github/workflows/test-build.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Test and build
```
Comment (averen) [inline .github/workflows/test-build.yml]: Rename to verify

File: .github/workflows/test-build.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+on: workflow_call
+
+jobs:
+  test:
```
Comment (averen) [inline .github/workflows/test-build.yml]: Testing is rather unnecessary as there are no tests

**Review by Legolog3 — APPROVED** [review]
mmmmmmmm looks good to me

**Pull request merged**